spec/models/participant_declaration/npq_spec.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe ParticipantDeclaration::NPQ, type: :model do
+  context "when declarations for a particular course are made" do
+    before do
+      create_list(:npq_participant_declaration, 20)
+    end
+
+    it "returns all records when not scoped" do
+      expect(described_class.all.count).to eq(20)
+    end
+
+    it "can retrieve by npq courses attached" do
+      expect(NPQCourse.identifiers.uniq.map { |course_identifier| described_class.for_course(course_identifier).count }.inject(:+)).to eq(20)
+    end
+  end
```

Comment:
I'm not sure this test has a lot of value, it's tested at a much higher level through the feature specs